### PR TITLE
Add tests for correct interpolations.

### PR DESCRIPTION
**Pull request by korsbo**
Test for correct interpolation and pullback behaviour.

### CODE REVIEW

**Comment by korsbo** [comment]
That seems to be working. Nice! 

So, then we need a better list of algorithms and to remove the annoying display(). Anything else?

File: test/ode/ode_interpolation_tests.jl [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+cb = ContinuousCallback(pullback_condition, pullback_affect!)
+
+algs = [Tsit5, Rosenbrock23, Rosenbrock32, Rodas5] ## Works for these
+# algs = subtypes(OrdinaryDiffEq.OrdinaryDiffEqAdaptiveAlgorithm)
+
+passed = fill(false, length(algs))
+
+
+for (i, alg) in enumerate(algs)
+    display("testing $alg")
```
Comment (korsbo) [inline test/ode/ode_interpolation_tests.jl]: This one should be deleted before a merge.

**Comment by ChrisRackauckas** [comment]
Thanks! This is a great test and will be very helpful down the line.

File: test/ode/ode_add_steps_tests.jl [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+lazy_alg = [BS5, Vern6, Vern7, Vern8, Vern9]
+
+nonstandard_interp_algs = union(algs,bad_algs,lazy_alg)
+
+passed = fill(false, 2length(algs))
+
+cur_itr = 0
+for inplace in [false,true], alg in nonstandard_interp_algs
+    prob = ODEProblem{inplace}(test_ode, [0.], tspan, [1.])
+    sol = solve(prob, alg(); callback=cb,dt=0.001)
```
[the remark this answers is not in the document]
Comment (ChrisRackauckas) [reply test/ode/ode_add_steps_tests.jl]: It only matters for fixed time step methods, but good point.

**Pull request merged**